marine/simple_marine.py

```diff
@@ -0,0 +1,120 @@
+from .marine import Marine
+from . import encap_consts
+from typing import Optional, List, Dict, Tuple
+
+marine_instance = None
+
+
+def init_instance(epan_auto_reset_count: Optional[int] = None) -> Marine:
+    global marine_instance
+
+    if marine_instance is None:
+        marine_instance = Marine(epan_auto_reset_count)
+    return marine_instance
+
+
+def filter_packet(
+    packet: bytes,
+    bpf: Optional[str] = None,
+    display_filter: Optional[str] = None,
+    encapsulation_type: int = encap_consts.ENCAP_ETHERNET,
+) -> bool:
+    """
+    Filters a packet with BPF and a Wireshark-style display filter.
+    At least one form of filtering is required.
+    By default the packet is parsed as an ethernet packet,
+    to view other possible encapsulation values view encap_consts.
+    """
+    return init_instance().filter(
+        packet=packet,
+        bpf=bpf,
+        display_filter=display_filter,
+        encapsulation_type=encapsulation_type,
+    )
+
+
+def parse_packet(
+    packet: bytes,
+    fields: Optional[List[str]] = None,
+    macros: Optional[Dict[str, List[str]]] = None,
+    encapsulation_type: int = encap_consts.ENCAP_ETHERNET,
```

Comment:
`None`

Reply:
Those are changes that are indeed related to the [encap-detection](https://github.com/tomlegkov/marine-python/pull/32) PR, which is why the typing changes are made there.